sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java

```diff
@@ -109,6 +109,13 @@ String getSessionToken(String collectionLink) {
     public String resolveGlobalSessionToken(RxDocumentServiceRequest request) {
         ConcurrentHashMap<String, ISessionToken> partitionKeyRangeIdToTokenMap = this.getPartitionKeyRangeIdToTokenMap(request);
         if (partitionKeyRangeIdToTokenMap != null) {
+            String partitionKeyRangeId = request.getHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
```

Comment:
Is this header present for all crud/query/readFeed ?

Reply:
Only checked for the ReadFeed, more precisely the ChangeFeedProcessor related request where  I could confirm that the header is set prior to adding the session token header.